### ISSUE TITLE
Extension errors, app goes crazy with 3.4.10 update

### ISSUE THREAD

**Issue by pcrockett** [standardnotes/desktop#623]
I updated to 3.4.10 on a Linux laptop when it came out. It seemed to be working fine.

After a few days, I updated my Windows desktop to 3.4.10 as well. On the first run of the new version, Standard Notes started misbehaving (see below for details) and I could never get it running after some troubleshooting steps. I gave up and put it away for a while.

Today I opened Standard Notes on my Linux laptop again (which was already on 3.4.10), and now it's behaving like it does on my Windows desktop. After some troubleshooting, I still can't get it to work.

### To reproduce

1. Start Standard Notes
2. Enter local passcode

After entering the local passcode, here's what happens:

1. I get a popup message saying, "An extension is trying to communicate with Standard Notes, but there is an error establishing a bridge. Please restart the app and try again."
2. Click OK. The popup shows a second time.
3. Click OK. The app starts flickering / stuttering / going crazy, and then the popup shows a third time.
4. Click OK. The popup stops reappearing, but the app continues going crazy and remains stuck in the middle of sync / decryption.

I force close the application and restart it, and I'm taken back to the local passcode page again. After entering the passcode, I continue getting the popups, the app continues going crazy.

### What do I mean by "going crazy"?

Three weird things happen in the UI:

1. My mouse cursor flickers back and forth rapidly between the normal "pointer arrow" and the "text input" cursor. Interestingly enough, I have seen this mouse flickering behavior before. I created issue #498, and once that issue was resolved (thanks, btw), the mouse flickering problem went away.
2. The title bar of whatever note happens to be displayed at the moment continually rearranges itself, adding whitespace and removing it, re-rendering, etc.
3. The status text at the bottom of the screen is stuck on "Decrypting 300 / 1327 items..."

### Troubleshooting steps

I saved Standard Notes output to a log file, and nothing seems terribly out of order:

```plaintext
extServer: Server started at http://127.0.0.1:45653/
08:36:46.768 › Checking for update
08:36:48.052 › Checking for update (already in progress)
08:36:49.105 › Update for version 3.4.10 is not available (latest version: 3.4.10, downgrade is disallowed).
PackageManager: received sync event for: Quick Tags (1.3.2) (deleted: false), FileSafe (2.0.10) (deleted: false), Quick Tags (1.3.2) (deleted: false), Simple Markdown Editor (1.3.6) (deleted: false), 2FA Manager (1.2.2) (deleted: false), CloudLink (1.2.1) (deleted: false), Extensions (undefined) (deleted: false), Batch Manager (undefined) (deleted: false), Dynamic (1.0.0) (deleted: false), Futura (1.2.2) (deleted: false), No Distraction (1.2.2) (deleted: false)
PackageManager: Checking for update for FileSafe
Latest: 2.0.10 | Installed: 2.0.10
PackageManager: Checking for update for No Distraction
Latest: 1.2.2 | Installed: 1.2.2
PackageManager: Checking for update for Quick Tags
Latest: 1.3.2 | Installed: 1.3.2
PackageManager: Checking for update for Dynamic
Latest: 1.0.0 | Installed: 1.0.0
PackageManager: Checking for update for Simple Markdown Editor
Latest: 1.3.6 | Installed: 1.3.6
PackageManager: Checking for update for 2FA Manager
Latest: 1.2.2 | Installed: 1.2.2
PackageManager: Checking for update for Quick Tags
Latest: 1.3.2 | Installed: 1.3.2
PackageManager: Checking for update for Futura
Latest: 1.2.2 | Installed: 1.2.2
```

I closed and relaunched, and instead of entering my local passcode, I clicked the link that causes my local data to be deleted and the app was reset. It took me to the default "fresh from scratch" UI and I was allowed to log in again. However after logging in again, I got the same behavior as described above. The log file I get now is a little bit longer:

```plaintext
extServer: Server started at http://127.0.0.1:45653/
08:39:39.233 › Checking for update
08:39:40.918 › Checking for update (already in progress)
08:39:40.961 › Update for version 3.4.10 is not available (latest version: 3.4.10, downgrade is disallowed).
PackageManager: received sync event for: Extensions (undefined) (deleted: undefined), Batch Manager (undefined) (deleted: undefined)
PackageManager: received sync event for: Extensions (undefined) (deleted: undefined), Batch Manager (undefined) (deleted: undefined)
archiveManager: Data backup successfully saved: 2020-10-25T07-40-10.785Z.txt
PackageManager: received sync event for: Extensions (undefined) (deleted: true)
PackageManager: Uninstalling Extensions
PackageManager: received sync event for: Extensions (undefined) (deleted: true)
PackageManager: Uninstalling Extensions
archiveManager: Data backup successfully saved: 2020-10-25T07-42-18.345Z.txt
PackageManager: received sync event for: Extensions (undefined) (deleted: undefined)
PackageManager: received sync event for: Batch Manager (undefined) (deleted: true)
PackageManager: Uninstalling Batch Manager
PackageManager: received sync event for: Batch Manager (undefined) (deleted: true)
PackageManager: Uninstalling Batch Manager
PackageManager: received sync event for: Batch Manager (undefined) (deleted: undefined)
PackageManager: received sync event for: Batch Manager (undefined) (deleted: false), Extensions (undefined) (deleted: false), CloudLink (1.2.1) (deleted: false)
PackageManager: received sync event for: Extensions (undefined) (deleted: true)
PackageManager: Uninstalling Extensions
PackageManager: received sync event for: Batch Manager (undefined) (deleted: true)
PackageManager: Uninstalling Batch Manager
PackageManager: received sync event for: Extensions (undefined) (deleted: true)
PackageManager: Uninstalling Extensions
PackageManager: received sync event for: Batch Manager (undefined) (deleted: true)
PackageManager: Uninstalling Batch Manager
PackageManager: received sync event for: Extensions (undefined) (deleted: true)
PackageManager: Uninstalling Extensions
PackageManager: received sync event for: Batch Manager (undefined) (deleted: true)
PackageManager: Uninstalling Batch Manager
PackageManager: received sync event for: 2FA Manager (1.2.2) (deleted: false), No Distraction (1.2.2) (deleted: false), Futura (1.2.2) (deleted: false)
PackageManager: Checking for update for 2FA Manager
Latest: 1.2.2 | Installed: 1.2.2
PackageManager: Checking for update for No Distraction
Latest: 1.2.2 | Installed: 1.2.2
PackageManager: Checking for update for Futura
Latest: 1.2.2 | Installed: 1.2.2
PackageManager: received sync event for: Futura (1.2.2) (deleted: false), Dynamic (1.0.0) (deleted: false)
PackageManager: Checking for update for Futura
Latest: 1.2.2 | Installed: 1.2.2
PackageManager: Checking for update for Dynamic
Latest: 1.0.0 | Installed: 1.0.0
PackageManager: received sync event for: Simple Markdown Editor (1.3.6) (deleted: false), Quick Tags (1.3.2) (deleted: false)
PackageManager: Checking for update for Simple Markdown Editor
Latest: 1.3.6 | Installed: 1.3.6
PackageManager: Checking for update for Quick Tags
Latest: 1.3.2 | Installed: 1.3.2
PackageManager: received sync event for: FileSafe (2.0.10) (deleted: false), Quick Tags (1.3.2) (deleted: false)
PackageManager: Checking for update for Quick Tags
Latest: 1.3.2 | Installed: 1.3.2
PackageManager: Checking for update for FileSafe
Latest: 2.0.10 | Installed: 2.0.10
```

Interestingly enough, now that I'm not entering a local passcode anymore, I no longer get the error popups. The sync / decrypt process starts right away, and the app just goes crazy.

**Comment by mobitar** [comment]
Can you try the 3.5 beta to see if that makes a difference? https://github.com/standardnotes/desktop/releases The latest betas are stable and pretty close to release.

**Comment by pcrockett** [comment]
Ooh, I like what I see in those release notes. I'd love to try it, but...

> Your data will only be readable by 3.5+ desktop clients, mobile clients, and web clients.

I don't see any beta releases for the mobile client yet, so does that mean the moment I run the beta desktop client, I'll lose the ability to use the mobile?

**Comment by pcrockett** [comment]
The issue seems to be resolved with 3.5.1-beta05. Pretty excited about those release notes, btw. Thanks!
